code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -4608,6 +4608,24 @@ var/global/list/chifir_doesnt_remove=list(
 	id = "gyro"
 	description = "Nyo ho ho~"
 
+/datum/reagent/drink/tea/gyro/on_mob_life(var/mob/living/M as mob)
+	if(!holder) return
+	if(!M) M = holder.my_atom
+	if(prob(30))
+		M.emote("spin")
+	var/prev_dir = M.dir
+	M.confused++
+	for(var/i in list(1,4,2,8,1,4,2,8,1,4,2,8,1,4,2,8))
```

Comment:
I'm worried that it might, unless on_mob_life itself is also spawned.
If you want to do that, tab the for loop and the healing under a spawn.